### PR TITLE
chore: add e2e tetss for js-cloud-server-sdk

### PR DESCRIPTION
**Pull request by suthar26**
- add e2e app and tests for the cloud server sdk, running using the build output

### CODE REVIEW

**Comment by suthar26** [comment]
* **#759** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/759?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @suthar26 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: e2e/js/js-cloud-server/playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+    use: {
+        baseURL,
+        /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+        trace: 'on-first-retry',
+        screenshot: 'only-on-failure',
+    },
+    /* Run your local dev server before starting the tests */
+    webServer: {
+        command: 'yarn e2e:js-esm:start',
```
Comment (jsalaber) [inline e2e/js/js-cloud-server/playwright.config.ts:36]: `js-cloud-server`

File: e2e/js/js-cloud-server/webpack.config.old.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin')
```
Comment (jsalaber) [inline e2e/js/js-cloud-server/webpack.config.old.js:1]: is this wrangler config even being used

File: e2e/js/js-cloud-server/project.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+    "name": "e2e-js-cloud-server",
+    "$schema": "../../../node_modules/nx/schemas/project-schema.json",
+    "sourceRoot": "e2e/nextjs/app-router/src",
+    "targets": {
+        "build": {
```
Comment (jsalaber) [inline e2e/js/js-cloud-server/project.json:6]: are these commands even needed? `build`, `serve`, `deploy`

Comment (suthar26) [reply e2e/js/js-cloud-server/project.json]: dont need build and deploy so removed those

File: e2e/js/js-cloud-server/playwright.config.ts (hunk 1 above)
Comment (suthar26) [reply e2e/js/js-cloud-server/playwright.config.ts]: removed, and dont need anymore

File: e2e/js/js-cloud-server/tests/js-cloud-server.spec.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+describe('js-cloud-server', () => {
+    it('should return all variables', async () => {
+        const response = await fetch('http://localhost:8787')
+        const text = await response.text()
+        expect(text).toContain('DevCycle Variables')
```
Comment (jsalaber) [inline e2e/js/js-cloud-server/tests/js-cloud-server.spec.ts:5]: should test the actual output from a project using the `.variable` function, use this project https://app.devcycle.com/o/org_U9F8YMaTChTEndWw

Comment (suthar26) [reply e2e/js/js-cloud-server/tests/js-cloud-server.spec.ts]: fixed it

**Review by jsalaber — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -24,7 +24,8 @@
         "e2e:nextjs-pages-router:start": "nx build nextjs && ./e2e/nextjs/pages-router/app/start.sh",
         "e2e:js-cdn:start": "nx run js-cdn:build-with-sdk && nx run js-cdn:serve",
         "e2e:js-esm:start": "nx build js && ./e2e/js/js-esm/start.sh",
-        "e2e:react:start": "nx build react && ./e2e/react/start.sh"
+        "e2e:react:start": "nx build react && ./e2e/react/start.sh",
+        "e2e:js-cloud-server:start": "nx build js-cloud-server-sdk && .e2e/js/js-cloud-server/start.sh"
```
Comment (jsalaber) [inline package.json:28]: `./e2e/`

File: .github/workflows/nx-affected-e2e.yml [hunk 6]
```diff
@@ -22,6 +22,10 @@ jobs:
             - name: Install Playwright Browsers
               shell: bash
               run: yarn playwright install --with-deps
+            - name: start cloud server worker
```
Comment (jsalaber) [inline .github/workflows/nx-affected-e2e.yml:25]: is this needed since there's a npm script that does this already?

Comment (suthar26) [reply .github/workflows/nx-affected-e2e.yml:25]: that script is not getting called and I was getting a fetch failed error so explicitly starting seems to be working

File: e2e/js/js-cloud-server/project.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        "serve": {
+            "executor": "nx:run-commands",
+            "options": {
+                "command": "yarn run -T wrangler dev",
+                "cwd": "examples/js-cloud-server/cloudflare-worker"
+            }
+        },
+        "e2e": {
+            "executor": "@nrwl/jest:jest",
+            "options": {
```
Comment (jsalaber) [inline e2e/js/js-cloud-server/project.json:15]: maybe add options to start the cloudflare worker here instead of in the yml file of the e2e tests

**Review by jsalaber — CHANGES_REQUESTED** [review]

File: e2e/js/js-cloud-server/project.json (hunk 7 above)
Comment (suthar26) [reply e2e/js/js-cloud-server/project.json:15]: I tried, and it does not seem to work, still getting `fetch failed` any reason against putting it in the workflow file?

Comment (jsalaber) [reply e2e/js/js-cloud-server/project.json:15]: same reason as the other comment, it's unnecessarily building the sdk and running the worker before it determines whether or not it even needs to

**Pull request merged**